docs/iam-usage.rst

```diff
@@ -0,0 +1,148 @@
+Using IAM For Users
+===================
+
+.. note:: This document assumes basic knowledge of Google Cloud IAM,
+   see `the docs <https://cloud.google.com/iam/docs/>`_ for details
+
+Resources that implement the IAM interface provide the following methods:
+
+Policy Editing Convenience Methods
+----------------------------------
+
+Common keyword arguments:
+
+- ``retries`` number of times to retry in the
+   even of an interrupted transaction, (not available in ``set_policy``)
+- ``version`` version number of the new policy. Not used for transactionality.
+   provided for end-user use. Left blank, will increment the version by 1.
+
+
+>>> resource.add_role(iam.user('alice@example.com'), iam.roles.OWNER.name)
```

Comment:
In the future Roles will be a full fledged resource, with the user able to create custom roles. A role has a name, a description, and a list of permissions it provides. The alternative to using .name I thought of was just making `__str__()` return the role's name.